File: frontend/src/types/content.ts
export type ContentStatus = 'draft' | 'queued' | 'publishing' | 'success' | 'partial_success' | 'failed';
export type Platform = 'fb' | 'ig' | 'x' | 'line';

export interface Content {
  id: string;
  title: string;
  master_caption: string;
  image_url: string | null;
  platforms: Platform[];
  publish_at: string | null;
  status: ContentStatus;
  fb_caption: string | null;
  ig_caption: string | null;
  x_caption: string | null;
  line_message: string | null;
  retry_count: number;
  last_error: string | null;
  created_at: string;
  updated_at: string;
}

export interface PublishLog {
  log_id: string;
  content_id: string;
  platform: Platform;
  status: 'success' | 'failed';
  external_post_id: string | null;
  response_summary: string | null;
  created_at: string;
}

export interface MonitorData {
  id: string;
  content_id: string;
  platform: Platform;
  external_post_id: string;
  likes: number;
  comments: number;
  shares: number;
  recent_replies: Reply[];
  fetched_at: string;
}

export interface Reply {
  author: string;
  text: string;
  created_at: string;
  platform: Platform;
}

export const STATUS_LABELS: Record<ContentStatus, string> = {
  draft: '草稿',
  queued: '排程中',
  publishing: '發佈中',
  success: '已發佈',
  partial_success: '部分成功',
  failed: '失敗',
};

export const PLATFORM_LABELS: Record<Platform, string> = {
  fb: 'Facebook',
  ig: 'Instagram',
  x: 'X',
  line: 'LINE',
};

export const PLATFORM_COLORS: Record<Platform, string> = {
  fb: 'bg-info',
  ig: 'bg-primary',
  x: 'bg-foreground',
  line: 'bg-success',
};

File: modules/module1-distributor/backend/app/models/content.py
import uuid
from datetime import datetime

from sqlalchemy import (
    JSON,
    DateTime,
    ForeignKey,
    Integer,
    String,
    Text,
    UniqueConstraint,
    func,
)
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.core.database import Base


class ContentQueue(Base):
    __tablename__ = "content_queue"

    id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    title: Mapped[str] = mapped_column(String(255))
    master_caption: Mapped[str] = mapped_column(Text)
    image_url: Mapped[str | None] = mapped_column(String(500), nullable=True)
    platforms: Mapped[str] = mapped_column(String(100))  # "fb,ig,x,line"
    publish_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True), nullable=True)
    status: Mapped[str] = mapped_column(String(20), default="draft")

    # Platform-specific captions
    fb_caption: Mapped[str | None] = mapped_column(Text, nullable=True)
    ig_caption: Mapped[str | None] = mapped_column(Text, nullable=True)
    x_caption: Mapped[str | None] = mapped_column(Text, nullable=True)
    line_message: Mapped[str | None] = mapped_column(Text, nullable=True)

    retry_count: Mapped[int] = mapped_column(Integer, default=0)
    last_error: Mapped[str | None] = mapped_column(Text, nullable=True)

    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now())
    updated_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now())

    publish_logs: Mapped[list["PublishLog"]] = relationship(back_populates="content", cascade="all, delete-orphan")
    monitor_data: Mapped[list["MonitorData"]] = relationship(back_populates="content", cascade="all, delete-orphan")


class PublishLog(Base):
    __tablename__ = "publish_logs"
    __table_args__ = (
        UniqueConstraint(
            "content_id",
            "platform",
            name="uq_publish_logs_content_platform",
        ),
    )

    log_id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    content_id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), ForeignKey("content_queue.id"))
    platform: Mapped[str] = mapped_column(String(20))
    status: Mapped[str] = mapped_column(String(20))
    external_post_id: Mapped[str | None] = mapped_column(String(255), nullable=True)
    response_summary: Mapped[str | None] = mapped_column(Text, nullable=True)
    workflow_execution_id: Mapped[str | None] = mapped_column(String(255), nullable=True)
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now())

    content: Mapped["ContentQueue"] = relationship(back_populates="publish_logs")


class MonitorData(Base):
    __tablename__ = "monitor_data"

    id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    content_id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), ForeignKey("content_queue.id"))
    platform: Mapped[str] = mapped_column(String(20))
    external_post_id: Mapped[str | None] = mapped_column(String(255), nullable=True)
    likes: Mapped[int] = mapped_column(Integer, default=0)
    comments: Mapped[int] = mapped_column(Integer, default=0)
    shares: Mapped[int] = mapped_column(Integer, default=0)
    recent_replies: Mapped[dict | None] = mapped_column(JSON, nullable=True)
    fetched_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now())

    content: Mapped["ContentQueue"] = relationship(back_populates="monitor_data")
